practiceapp/src/test/java/com/practiceapp/practiceapp/service/ProfileServiceTest.java

```diff
@@ -0,0 +1,50 @@
+package com.practiceapp.practiceapp.service;
+
+import com.practiceapp.practiceapp.entity.Profile;
+import com.practiceapp.practiceapp.repository.ProfileRepository;
+import com.practiceapp.practiceapp.utils.PicturesApi;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@WebMvcTest(ProfileService.class)
+public class ProfileServiceTest {
+    @Autowired
+    private ProfileService profileService;
+
+    @MockBean
+    private ProfileRepository profileRepository;
+
+
+    private Profile profile;
+
+    @BeforeEach
+    void setUp() {
+        profile = new Profile();
+        profile.setId("1222");
```

Comment:
```suggestion
```

You can delete this to avoid errors caused by ObjectId. I think It will not affect the quality of the test.

Reply:
I implemented the changes you mentioned. I also add more features.
SetRandomPic and getProfile APIs are not taking profile id anymore. It takes username because we though it might be easier for the user to give username instead of an id. I also add some documentation parts. 
Please check

Reply:
You are right. Using username instead of ID is more user-friendly. Thanks for the improvement.